include/hidet/cpu/bfloat16.h

```diff
@@ -0,0 +1,686 @@
+      ----------------------------------------------------------------------
+      Copyright © 2005-2020 Rich Felker, et al.
+      Permission is hereby granted, free of charge, to any person obtaining
+      a copy of this software and associated documentation files (the
+      "Software"), to deal in the Software without restriction, including
+      without limitation the rights to use, copy, modify, merge, publish,
+      distribute, sublicense, and/or sell copies of the Software, and to
+      permit persons to whom the Software is furnished to do so, subject to
+      the following conditions:
+      The above copyright notice and this permission notice shall be
+      included in all copies or substantial portions of the Software.
+      THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+      ----------------------------------------------------------------------
+     */
+    inline BFloat16 nextafter(
+        BFloat16 from,
+        BFloat16 to)
+    {
+        // Reference:
+        // https://git.musl-libc.org/cgit/musl/tree/src/math/nextafter.c
+        using int_repr_t = uint16_t;
+        using float_t = BFloat16;
+        constexpr uint8_t bits = 16;
+        union
+        {
+            float_t f;
+            int_repr_t i;
+        } ufrom = {from}, uto = {to};
+
+        // get a mask to get the sign bit i.e. MSB
+        int_repr_t sign_mask = int_repr_t{1} << (bits - 1);
+
+        // short-circuit: if either is NaN, return NaN
+        if (from != from || to != to)
+        {
+            return from + to;
+        }
+
+        // short-circuit: if they are exactly the same.
+        if (ufrom.i == uto.i)
+        {
+            return from;
+        }
+
+        // mask the sign-bit to zero i.e. positive
+        // equivalent to abs(x)
+        int_repr_t abs_from = ufrom.i & ~sign_mask;
+        int_repr_t abs_to = uto.i & ~sign_mask;
+        if (abs_from == 0)
+        {
+            // if both are zero but with different sign,
+            // preserve the sign of `to`.
+            if (abs_to == 0)
+            {
+                return to;
+            }
+            // smallest subnormal with sign of `to`.
+            ufrom.i = (uto.i & sign_mask) | int_repr_t{1};
+            return ufrom.f;
+        }
+
+        // if abs(from) > abs(to) or sign(from) != sign(to)
+        if (abs_from > abs_to || ((ufrom.i ^ uto.i) & sign_mask))
+        {
+            ufrom.i--;
+        }
+        else
+        {
+            ufrom.i++;
+        }
+
+        return ufrom.f;
+    }
```

Comment:
Anywhere this function is used?

Reply:
It's not used in hidet for now, I will remove it from the header.